src/stratigraphy/util/plot_utils.py

```diff
@@ -15,15 +20,20 @@ def _draw_lines(open_cv_img, lines, scale_factor=1):
             (5 * line_count) % 255,
             (10 * line_count) % 255,
         )
-        cv2.line(
-            open_cv_img,
-            np.dot(scale_factor, line.start.tuple),
-            np.dot(scale_factor, line.end.tuple),
-            color,
-            1,
-        )
-        cv2.circle(open_cv_img, np.dot(scale_factor, line.start.tuple), radius=1, color=(0, 0, 255), thickness=-1)
-        cv2.circle(open_cv_img, np.dot(scale_factor, line.end.tuple), radius=1, color=(0, 0, 255), thickness=-1)
+        try:
+            cv2.line(
+                open_cv_img,
+                np.dot(scale_factor, line.start.tuple),
+                np.dot(scale_factor, line.end.tuple),
+                color,
+                1,
+            )
+            cv2.circle(open_cv_img, np.dot(scale_factor, line.start.tuple), radius=1, color=(0, 0, 255), thickness=-1)
+            cv2.circle(open_cv_img, np.dot(scale_factor, line.end.tuple), radius=1, color=(0, 0, 255), thickness=-1)
+
+        except cv2.error as e:
+            logging.warning(f"Error drawing line. Exception: {e}. Skipping to draw the line.")
```

Comment:
Happened for nan's. E.g. for when we attempt to merge vertical lines.

No warning is thrown now, as we can deal with all the cornercases present in our data.